File: sentinelai-frontend/src/types/alert.ts
export interface Alert {
  id: string;
  title: string;
  description: string | null;
  severity: 'critical' | 'high' | 'medium' | 'low' | 'informational';
  status: 'open' | 'acknowledged' | 'investigating' | 'resolved' | 'false_positive';
  source: string | null;
  source_ip: string | null;
  destination_ip: string | null;
  source_port: number | null;
  destination_port: number | null;
  protocol: string | null;
  mitre_technique_id: string | null;
  mitre_tactic: string | null;
  rule_id: string | null;
  rule_name: string | null;
  score: number;
  raw_data: Record<string, unknown> | null;
  enriched_data: Record<string, unknown> | null;
  tags: string[] | null;
  asset_ids: string[] | null;
  country: string | null;
  city: string | null;
  correlation_group_id: string | null;
  recommendation: string | null;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  resolved_by: string | null;
  resolved_at: string | null;
  incident_id: string | null;
  created_at: string;
  updated_at: string;
  created_by: string | null;
}

export interface AlertListResponse {
  items: Alert[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface AlertStatsResponse {
  total: number;
  by_severity: Record<string, number>;
  by_status: Record<string, number>;
  by_rule: Record<string, number>;
  top_source_ips: Array<{ ip: string; count: number }>;
  avg_score: number;
  recent_trend: Array<{ date: string; count: number }>;
}

File: sentinelai-backend/app/schemas/alert.py
from datetime import datetime
from typing import Any

from app.schemas.base import BaseSchema


class AlertResponse(BaseSchema):
    id: str
    title: str
    description: str | None = None
    severity: str
    status: str
    source: str | None = None
    source_ip: str | None = None
    destination_ip: str | None = None
    source_port: int | None = None
    destination_port: int | None = None
    protocol: str | None = None
    mitre_technique_id: str | None = None
    mitre_tactic: str | None = None
    rule_id: str | None = None
    rule_name: str | None = None
    score: int = 0
    raw_data: dict[str, Any] | None = None
    enriched_data: dict[str, Any] | None = None
    tags: list[str] | None = None
    asset_ids: list[str] | None = None
    country: str | None = None
    city: str | None = None
    correlation_group_id: str | None = None
    recommendation: str | None = None
    acknowledged_by: str | None = None
    acknowledged_at: datetime | None = None
    resolved_by: str | None = None
    resolved_at: datetime | None = None
    incident_id: str | None = None
    created_at: datetime
    updated_at: datetime
    created_by: str | None = None


class AlertUpdate(BaseSchema):
    status: str | None = None
    acknowledged_by: str | None = None
    resolved_by: str | None = None
    incident_id: str | None = None


class AlertStatsResponse(BaseSchema):
    total: int
    by_severity: dict[str, int]
    by_status: dict[str, int]
    by_rule: dict[str, int]
    top_source_ips: list[dict[str, Any]]
    avg_score: float
    recent_trend: list[dict[str, Any]]


class AlertListResponse(BaseSchema):
    items: list[AlertResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
